tests/test-cases/Group23-VIC-Machine-Service/23-02-VCH-List.md

```diff
@@ -10,17 +10,72 @@ To verify vic-machine-server can return a list of VCHs including the expected in
 # Environment:
 This test requires a vSphere system where VCHs can be deployed 
 
-# Test Steps:
-1. Deploy a VCH
-2. Get a list of all VCHs
-3. Get a list of all VCHs in the test datacenter
-4. Attempt to list all VCHs in an invalid datacenter
-5. Attempt to list all VCHs in an invalid compute resource
-6. Attempt to list all VCHs in an invalid datacenter and compute resource
-
-# Expected Outcome:
-* The results of 2-3 should contain the VCH created in 1, with the correct ID.
-* The requests in 4-6 should result in an appropriate error message.
-
-# Possible Problems:
-None known
+**Note:** Multiple versions of tests must be implemented:
+ * With and without datacenter-scoped URLs.
+ * Using both username/password- and session-based authentication.
+
+It should not be necessary to implement all four combinations for every test case, but full coverage should be provided for at least one of the basic test cases.
+
+
+Basic Validation
+----------------
+
+These tests are intended to verify the feature at a basic level.
+
+###  1. Without creating any VCHs, verify that an empty list is properly returned
+
+(Not yet implemented)
+
+###  2. Create a VCH and verify it is returned in the list, with correct information
+
+###  3. Create two VCHs and verify both are returned in the list, with correct information for each
+
+(Not yet implemented)
+
+###  4. Create 64 VCHs and verify all are returned in the list, with correct information for each
+
+(Not yet implemented)
+
+**Note:** This case may be better implemented as a nightly test, or similar; it is too onerous to run as a part of CI.
+
+
+Negative Cases
+--------------
+
+These tests are intended to verify the behavior in various failure cases. We must gracefully handle invalid input and unexpected user behavior.
+
+###  1. Attempt to list the VCHs in an invalid datacenter
+
+###  2. Attempt to list the VCHs in an invalid compute resource
+
+###  3. Attempt to list the VCHs in an invalid compute resource and datacenter
+
+###  4. Ensure that a VCH in one datacenter is not returned when listing the VCHs in another datacenter
+
+(Not yet implemented)
+
+
+Interoperability
+----------------
+
+These tests are intended to verify that the API and CLI can coexist without issue.
+
+###  1. Create a VCH using the CLI and verify that it is correctly included when listing via the API
+
+
+Concurrency
+-----------
+
+These tests are intended to verify that the API behaves as expected when performing concurrent operations.
+
+###  1. List VCHs while one is being created
```

Comment:
Do you think if it's a good idea for us to have separate test files called `concurrency.md` and `concurrency.robot` under `test-cases/Group23-VIC-Machine-Service` for testing concurrent API operations only? So we can put all of the concurrency related testing, including configure and upgrade/rollback into the same test file. 
Similarly is it a good idea to also have `workflow.md` and `workflow.robot` for common user workflows of using API, so we can unify all the workflow based API tests for different methods together?

Besides, implementation wise, do we have an exiting pattern to do concurrent testing with robot? Or is it easier to test it on nightly

Reply:
I think we can reorganize the tests as we implement them. We should find a pattern that meets the expectations of our CI and other developers, while allowing us to have reasonably-sized suites.